addons/nuxeo-platform-document-routing/nuxeo-routing-core/src/main/java/org/nuxeo/ecm/platform/routing/core/api/operation/StartWorkflowOperation.java

```diff
@@ -97,13 +98,19 @@ public DocumentModel run(DocumentModel doc) {
     }
 
     protected void startNewInstance(List<String> ids) {
-        Map<String, Serializable> vars = new HashMap<>();
+        Map<String, Serializable> vars = null;
         if (variables != null) {
-            for (Entry<String, String> entry : variables.entrySet()) {
-                vars.put(entry.getKey(), entry.getValue());
+            if (variables instanceof DataModelProperties) {
+                vars = ((DataModelProperties) variables).getMap();
+                vars.put(DocumentRoutingConstants._MAP_VAR_FORMAT_JSON, false);
+            } else {
+                vars = new HashMap<>();
+                for (Entry<String, String> entry : variables.entrySet()) {
+                    vars.put(entry.getKey(), entry.getValue());
+                }
```

Comment:
We can improve on the old code and just do `vars = new HashMap<>(variables)` here.